aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -757,7 +758,7 @@ Tensor scaled_dot_product_attention(
       const auto all_contiguous = query_.is_contiguous() && key.is_contiguous() && value.is_contiguous();
       if (query_device_type == DeviceType::MPS && dropout_p == 0.0
           && !(GradMode::is_enabled() && any_inputs_require_grad)
-          && all_contiguous
+          && (all_contiguous || mps::is_macos_13_or_newer(mps::MacOSVersion::MACOS_VER_15_0_PLUS))
```

Comment:
goes back to my last comment, that this codepath doesnt is kinda anti pattern to with how we dispatch to different backends

but that is somewhat beyond the scope of this PR

Reply:
@malfet there is a specific pattern in the SDPA op which is not optimized and we don't recognize the pattern for it and it causes the slow down. I don't think we need to disable it for all cases. cc @jhavukainen

Reply:
@kulinseth this simply enables call to SDPA op for-non contiguous tensors on MacOS15, as well are using strided API